### PR TITLE
Tear down

### PR DESCRIPTION
**Pull request by prat-gpt**
Tear down using: repayDebt, redeemLenderLP, validateEmpty methods and create a tear down modifier.

### CODE REVIEW

File: src/_test/ERC20Pool/ERC20DSTestPlus.sol
Comment (EdNoepel) [inline src/_test/ERC20Pool/ERC20DSTestPlus.sol]: Curious whether another `_pool.lenderInfo` would give a 0 `lenderLpBalance`.  If so, yet this assertion fails, it would mean there's a bug calculating amount of LPB redeemed in either `removeAllQuoteToken` or `removeAllCollateral`.

File: src/_test/utils/DSTestPlus.sol
Comment (EdNoepel) [inline src/_test/utils/DSTestPlus.sol]: You could use an [OpenZeppelin EnumerableSet](https://docs.openzeppelin.com/contracts/4.x/api/utils#EnumerableSet) for these, so you could add and look up with a single line of code.  Downside is impact on contract size, which shouldn't matter for unit tests.  Thoughts?

**Review by EdNoepel — COMMENTED** [review]
Thanks for adding `lpsToCollateral`.  Looks pretty good.

File: src/_test/utils/DSTestPlus.sol
Comment (prat-gpt) [reply src/_test/utils/DSTestPlus.sol]: Yes these are unit tests so won't impact the contract size. Agree with you, will use EnumerableSet for these.

File: src/_test/ERC20Pool/ERC20DSTestPlus.sol
Comment (prat-gpt) [reply src/_test/ERC20Pool/ERC20DSTestPlus.sol]: I checked this, `lendeLpBalance` is coming out as 0 from another `_pool.lenderInfo`

File: src/_test/ERC20Pool/ERC20DSTestPlus.sol
Comment (grandizzy) [inline src/_test/ERC20Pool/ERC20DSTestPlus.sol:138]: I think here beside validating buckets are empty we should assert also pool state and make sure pool is empty too (similar with `_assertPool` maybe?)

**Review by EdNoepel — APPROVED** [review]
Looks good.  Once `arbTake` and `depositTake` are implemented, we'll need to handle defragmentation of available collateral across the book, or just not use the `teardown` modifier on those tests.

Please await another review before merging, since I have a commit in here.

File: src/_test/ERC20Pool/ERC20DSTestPlus.sol
Comment (prat-gpt) [reply src/_test/ERC20Pool/ERC20DSTestPlus.sol:138]: I have added a commit that adds the pool asserts.

**Review by grandizzy — APPROVED** [review]
Looks good! style needs some love, like `if(currentDebt > 0){` should be `if (currentDebt > 0) {` or `for( uint j = 0; j < indexes.length() ; j++ ){` should be `for (uint j = 0; j < indexes.length() ; j++) {` and `external tearDown {` but feel free to merge as it is and we can fix style later

**Review by EdNoepel — APPROVED** [review]
LG;TM

**Pull request merged**
